include/exadg/solvers_and_preconditioners/preconditioners/preconditioner_amg.h

```diff
@@ -96,8 +162,11 @@ class PreconditionerML : public PreconditionerBase<double>
   dealii::TrilinosWrappers::PreconditionAMG amg;
 
 public:
-  PreconditionerML(Operator const & op, bool const initialize, MLData ml_data = MLData())
-    : pde_operator(op), ml_data(ml_data)
+  PreconditionerML(Operator const &     op,
+                   bool const           initialize,
+                   MLOperatorType const ml_operator_type,
```

Comment:
I think we should avoid copying the data and the related if-defs to set the operator type. Making `MLOperatorType` a constructor argument of `PreconditionerAMG` has the problem that in the multigrid preconditioner (and coarse grid solver) implementation we do not know the operator type, since this is generic code so far. So we either have to extend the interface of such generic multigrid classes, or instead query the required information from the PDE operator. Regarding the latter, different options are possible in my opinion:

i) pde_operator->get_pde_operator_type()
ii) pde_operator->get_rigid_body_modes()

Reply:
I tend to prefer ii), but then it would be good if we do not need `get_pde_operator_type()` in addition. Currently, we do not compute rigid body modes for Laplace and use somewhat different code paths instead. So my question is here probably again why we have the two options of AMG-initialization with `ml_data` or `parameter_list`.

Reply:
I unified the call now for an operator by setting up a `Teuchos:.ParameterList` in any case. This might impact some performance results (I guess only mildly), since setting up the `PreconditionerML` through `ml_data` did not per default allow repartitioning afaik.